src/GobanSocket.ts

```diff
@@ -115,22 +123,36 @@ export class OGSSocket extends EventEmitter<Events> {
         return this.socket.readyState === WebSocket.OPEN;
     }
 
+    public authenticate(authentication: DataArgument<ClientToServer["authenticate"]>): void {
+        this.authentication = authentication;
+        this.send("authenticate" as keyof SendProtocol, authentication as any);
+    }
+
     private sendAuthentication(): void {
-        //console.log("TODO: Implement authentication");
+        if (this.authentication) {
+            this.send("authenticate" as keyof SendProtocol, this.authentication as any);
```

Comment:
I think we can get rid of some type assertions by telling TS that `SendProtocol` extends `{"authenticate": ..., "net/ping": ...}`.  I'll open a pull request trying this out!

Reply:
Yeah good call, and thanks for the PR :)